text/0122-sdk-hub-scope-merge.md

```diff
@@ -0,0 +1,475 @@
+While it is empty it does nothing anyhow.
+
+If you cannot really fork a scope per execution context (e.g. in Browser JS), it's up to the SDK to implement the desired behavior in a similar way. The goal is to have a similar UX.
+
+## What should be called from top level methods?
+
+Top level APIs that set something should generally interact with the isolation scope:
+
+```js
+Sentry.setTag();
+Sentry.setUser();
+Sentry.addBreadcrumb();
+// ...
+```
+
+Top level APIs to capture events etc. interact with the current scope:
+
+```js
+Sentry.captureException();
+Sentry.captureMessage();
+// ...
+```
+
+SDKs _may_ also add an option to the client to opt-in to put breadcrumbs on the global scope instead (e.g. for mobile or scenarios where you always want breadcrumbs to be global).
+
+Note that since most data should generally be put on the isolation scope, we should not generally have to copy data that much on frequent scope forking. Since the isolation scope is rarely forked, most data does not need to be copied that frequently.
+
+## Should users care about global / isolation scopes?
+
+The goal of this RFC is that 95% of users do not need to think about these concepts at all.
+In most cases, users should simply use the global methods `Sentry.xyz()` which should do the right thing.
+
+Directly interacting with the isolation scope should only be necessary for users manually instrumenting an unsupported framework or similar. This (esp. setting up a new isolation scope) can be documented in a dedicated new docs page.
+
+Accessing the current scope `getCurrentScope()` should also rarely be necessary, and be de-emphazized in the docs. Again, just use the global methods which should do the correct thing.
+
+Accessing the global scope should be documented, and is the recommended way when you want to set global data in an async way. This can be encapsulated in a dedicated new docs page.
+
+We should also add a new docs page explaining the types of scope conceptually.
+
+## Should users care about Clients?
+
+Generally speaking, for most regular use cases the client should be mostly hidden away from users.
+Users should just call `Sentry.init()`, which will setup a client under the hood. Users should generally only interact with scopes, and we should keep clients out of most public facing APIs.
+
+The client is only there to allow an escape hatch when users need to do more complex/special things, like isolating Sentry instances or multiplexing. So client APIs should be designed to _allow_ to do things that cannot be done via `Sentry.init()`, but our main focus should be on making the default experience easy to understand, which includes that users should not have to care about the concept of clients by default.
+
+## What about other Hub references?
+
+While the Hub is mainly exposed via `getCurrentHub()`, it is also used as argument or similar in many places.
```

Comment:
Wouldn't this need to access all three scopes (global, isolation and current) instead of the single hub?